user-operator/src/main/java/io/strimzi/operator/user/operator/ScramShaCredentials.java

```diff
@@ -76,8 +73,8 @@ public void delete(String username) {
 
         if (data != null)   {
             log.debug("Deleting {} credentials for user {}", mechanism.mechanismName(), username);
-            byte[] deletedJson = deleteUserJson(data);
-            if (jsonConfigEmpty(deletedJson)) {
+            JsonObject deletedJson = deleteUserJson(data);
+            if (configJsonIsEmpty(deletedJson)) {
                 zkClient.deleteRecursive("/config/users/" + username);
             } else {
                 zkClient.writeData("/config/users/" + username, deletedJson);
```

Comment:
I think we need to serialize here.

Reply:
WDYM?

Reply:
Are you telling me that zkClient knows how to write a `JsonObject`? Maybe it does, by using `toString` and converting that to UTF-8 bytes. But maybe it doesn't.

Reply:
It seems the serialization is happening https://github.com/sgroschupf/zkclient/blob/master/src/main/java/org/I0Itec/zkclient/ZkClient.java#L1147

Reply:
But what does that actually _do_? It would be better to do it ourselves here, that way:

1. We know it's doing the right thing, and always will.
2. We potentially avoid a pointless intermediate String being created.